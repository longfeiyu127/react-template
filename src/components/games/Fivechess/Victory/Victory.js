import React from 'react';
import './Victory.less';

export default function Victory(props) {
  // console.log(props)
  // eslint-disable-next-line
  const { winner, reset } = props;
  return (
    <div className="g-fic-victory">
      <p className="victory">victoryï¼š {winner}</p>
      <p className="result" onClick={reset}>
        Again
      </p>
    </div>
  );
}
