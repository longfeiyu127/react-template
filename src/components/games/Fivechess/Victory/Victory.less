@import '../../../../assets/styles/theme.less';
.g-fic-victory{
  // background: white;
  font-size: 0.34rem;
  position: absolute;
  width: 60%;
  height: 1.3rem;
  text-align: center;
  top: 50%;
  left: 50%;
  margin-left: -30%;
  margin-top: -50%;
  font-family: 'TheHistoriaDemo';
  .victory{
    .victory;
  }
  .result{
    color: @warning-button-fill;
  }
  .result:hover{
    color: @warning-button-fill-tap;
  }
}