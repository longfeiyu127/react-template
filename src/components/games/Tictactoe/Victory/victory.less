.g-tic-victory{
  // background: white;
  font-size: 0.12rem;
  color: red;
  position: absolute;
  width: 60%;
  height: 1.3rem;
  text-align: center;
  top: 50%;
  left: 50%;
  margin-left: -30%;
  margin-top: -50%;
}