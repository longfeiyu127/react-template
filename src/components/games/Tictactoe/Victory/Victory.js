import React from 'react';
import { connect } from 'react-redux';
import './victory.less';

function TictactoeVictory(props) {
  /* eslint-disable */
  const { xIsNext, reset } = props;
  /* eslint-enable */
  const winner = xIsNext ? 'playerB' : 'playerA';
  return (
    <div className="g-tic-victory">
      <p>{winner} 胜利了</p>
      <p onClick={reset}>再来</p>
    </div>
  );
}

const mapStateToProps = state => ({
  xIsNext: state.Tictactoe.xIsNext
});

const mapDispatchToProps = dispatch => ({
  reset: dispatch.Tictactoe.setReset
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TictactoeVictory);
