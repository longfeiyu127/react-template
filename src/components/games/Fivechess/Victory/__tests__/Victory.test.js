import React from 'react';
import renderer from 'react-test-renderer';
import Victory from '../Victory';

describe('<Victory />', () => {
  it('component render', () => {
    let victory = renderer.create(<Victory />).toJSON();
    expect(victory).toMatchSnapshot();

    victory = renderer.create(<Victory winner="ç™½" />).toJSON();
    expect(victory).toMatchSnapshot();
  });
});
