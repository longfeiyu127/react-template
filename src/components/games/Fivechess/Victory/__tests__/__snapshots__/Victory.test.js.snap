// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Victory /> component render 1`] = `
<div
  className="g-fic-victory"
>
  <p
    className="victory"
  >
    victory： 
  </p>
  <p
    className="result"
  >
    Again
  </p>
</div>
`;

exports[`<Victory /> component render 2`] = `
<div
  className="g-fic-victory"
>
  <p
    className="victory"
  >
    victory： 
    白
  </p>
  <p
    className="result"
  >
    Again
  </p>
</div>
`;
